### PR TITLE
different attention map for each element of feature dimension

### PR DESCRIPTION
**Pull request by lucidrains**

### CODE REVIEW

File: point_transformer_pytorch/point_transformer_pytorch.py [hunk 1]
```diff
@@ -47,8 +47,8 @@ def forward(self, x, pos):
         v = v + rel_pos_emb
 
         # attention
-        attn = sim.softmax(dim = -1)
+        attn = sim.softmax(dim = -2)
```
Comment (Liu-Feng) [inline point_transformer_pytorch/point_transformer_pytorch.py:50]: I cannot confirm which axis that the softmax conduct on.

Comment (lucidrains) [reply point_transformer_pytorch/point_transformer_pytorch.py:50]: i'm pretty confident this is the right axis `j` - sim is `(batch x query rows x query-key similarities per row x feature dimension)`

Comment (lucidrains) [reply point_transformer_pytorch/point_transformer_pytorch.py:50]: i'm just not sure if the attention MLP does not reduce to one attention map that aggregates across all feature value vectors. if someone else can confirm this is indeed the case, i'll merge and release the corrected version

Comment (Liu-Feng) [reply point_transformer_pytorch/point_transformer_pytorch.py:50]: In my opinion, the point transformer in paper is used to informtion exchange or interaction, other than aggregation. The maxpooing in Pointnet++ is assigned to feature aggregation which aggregate many points feature to one point feature. In point transformer layer, the point number of outputs is same as the inputs.

Comment (lucidrains) [reply point_transformer_pytorch/point_transformer_pytorch.py:50]: ok, i'll go by your word for now :)

**Pull request merged**